crates/red_knot_python_semantic/src/types.rs

```diff
@@ -280,7 +284,7 @@ impl<'db> Type<'db> {
                 Type::Unknown
             }
             Type::BooleanLiteral(_) => Type::Unknown,
-            Type::StringLiteral(_) => {
+            Type::StringLiteral(_) | Type::LiteralString => {
                 // TODO defer to Type::Instance(<str from typeshed>).member
                 Type::Unknown
```

Comment:
Let's actually separate these cases and take into account @AlexWaygood's comment at https://github.com/astral-sh/ruff/pull/13113#discussion_r1731661129 -- that the `member` implementation for `StringLiteral` should in many cases be `LiteralString`, not based on the type annotations on `builtins.str` in typeshed.